### PR TITLE
FOUR-29250 End Event – External URL with Mustache Support

### PR DESCRIPTION
**Pull request by gproly**
Description:
feat(end event): support Mustache in external URL for element destination

- Resolve Mustache expressions in end event "External URL" when the token reaches the end event, using the same context as scripts/screens (APP_URL, _request, _user, process variables).
- Add getElementDestinationMustacheContext() to build context via DataManager with fallback; normalize to plain array for Mustache.
- Add resolveElementDestinationUrl() to decode HTML entities and render URL template with MustacheExpressionEvaluator. FEEL is not supported.
- Apply resolution only for externalURL type; conditional redirect URLs also go through Mustache when destination is external URL.
- Harden getElementDestinationAttribute(): ensure conditionalRedirectProp and elementDestinationProp are never null (json_decode ?? [], pass ?? []).
- Modeler: allow URL validation when string contains {{ (Mustache); update helper and error copy to document _request.id, _user.id, process vars. 
 

Related tickets:
https://processmaker.atlassian.net/browse/FOUR-29250

ci:deploy
ci:modeler:feature/FOUR-29250

---

> [!NOTE]
> **Medium Risk**
> Changes runtime redirect URL generation and conditional redirect evaluation, which can alter navigation behavior and expose more process/user data via templating if misconfigured. Added normalization and new tests reduce risk but the redirect path is user-facing and sensitive to edge cases.
> 
> **Overview**
> End event `externalURL`/`customDashboard` element destinations now support Mustache placeholders (e.g. `{{APP_URL}}`, `{{_request.id}}`, process variables) by building a DataManager-based context (with fallback), HTML-entity decoding the template, and rendering via `MustacheExpressionEvaluator` while stripping `remember_token`.
> 
> Conditional redirect evaluation now normalizes numeric strings in process data to numbers before FEEL execution to avoid mismatched comparisons (e.g. `"0"` vs `0`). A new unit test suite covers Mustache context construction, URL rendering (request/user/process vars), entity decoding, and token/user edge cases.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 2cbf37b73d0dba9f988f70ba713d39a374210486. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

**Comment by vladyrichter** [comment]
QA server K8S was successfully deployed https://ci-4a306d1d3c.engk8s.processmaker.net

File: ProcessMaker/Models/ProcessRequestToken.php [hunk 1]
```diff
@@ -1440,6 +1442,49 @@ public function reassign($toUserId, User $requestingUser, $comments = '')
...
+     * Build context for Mustache (end event external URL). Same as scripts/screens: _user, _request, process data, APP_URL.
+     */
+    private function getElementDestinationMustacheContext(): array
+    {
+        try {
+            $context = (new DataManager())->getData($this);
+        } catch (Throwable $e) {
+            $request = $this->processRequest;
+            $context = array_merge($request->data ?? [], $request ? (new DataManager())->updateRequestMagicVariable([], $request) : []);
+            $user = $this->user ?? \Illuminate\Support\Facades\Auth::user();
```
Comment (devmiguelangel) [inline ProcessMaker/Models/ProcessRequestToken.php]: use the auth() helper instead of \Illuminate\Support\Facades\Auth::user()
```suggestion
            $user = $this->user ?? auth()->user();

```

File: ProcessMaker/Models/ProcessRequestToken.php [hunk 2]
```diff
@@ -1440,6 +1442,49 @@ public function reassign($toUserId, User $requestingUser, $comments = '')
...
     }
 
+    /**
+     * Build context for Mustache (end event external URL). Same as scripts/screens: _user, _request, process data, APP_URL.
+     */
+    private function getElementDestinationMustacheContext(): array
+    {
+        try {
+            $context = (new DataManager())->getData($this);
+        } catch (Throwable $e) {
```
Comment (devmiguelangel) [inline ProcessMaker/Models/ProcessRequestToken.php:1452]: this catch block catches everything, consider to add a log to avoid a silent exception e.g.
```suggestion
        } catch (Throwable $e) {
            Log::warning('Failed to load Mustache context via DataManager, falling back to request data', [
                'token_id' => $this->id,
                'error' => $e->getMessage(),
            ]);
```

File: ProcessMaker/Models/ProcessRequestToken.php [hunk 3]
```diff
@@ -1440,6 +1442,49 @@ public function reassign($toUserId, User $requestingUser, $comments = '')
...
+                $userData = $user->attributesToArray();
+                unset($userData['remember_token']);
+                $context['_user'] = $userData;
+            }
+        }
+
+        $context['APP_URL'] = config('app.url');
+
+        // Normalize to plain arrays/scalars so Mustache resolves all keys (common PHP idiom)
+        $normalized = json_decode(json_encode($context), true);
```
Comment (devmiguelangel) [inline ProcessMaker/Models/ProcessRequestToken.php]: Add JSON_THROW_ON_ERROR to surface encoding failures
```suggestion
        $json = json_encode($context, JSON_THROW_ON_ERROR);
        $normalized = json_decode($json, true, 512, JSON_THROW_ON_ERROR);
```

**Review by devmiguelangel — CHANGES_REQUESTED** [review]

**Comment by vladyrichter** [comment]
QA server K8S was successfully deployed https://ci-4a306d1d3c.engk8s.processmaker.net

**Comment by vladyrichter** [comment]
QA server K8S was successfully deployed https://ci-4a306d1d3c.engk8s.processmaker.net

**Comment by Kookster310** [comment]
QA server K8S was successfully deployed https://ci-4a306d1d3c.engk8s.processmaker.net

**Review by devmiguelangel — APPROVED** [review]

**Review by rodriquelca — APPROVED** [review]
